test/powershell/engine/PropertyAccessor.Tests.ps1

```diff
@@ -0,0 +1,96 @@
+#
+# Functional tests to verify basic conditions for IO to the PowerShellProperties.json files
+# The properties files are supported on non-Windows OSes, but the tests are specific to 
+# Windows so that file IO can be verified using supported cmdlets.
+# 
+
+try {
+
+    #skip all tests on non-windows platform
+    $originalDefaultParameterValues = $PSDefaultParameterValues.Clone()
```

Comment:
Thanks for the link the discussion. I think it depends on how many `Describe` sections you expect to be in a file. If there is just one `Describe`, then it's cleaner to have it in `BeforeAlll` and `AfterAll`. But I'm fine keep the try/finally as it's possible that we will add more `Describe` sections later.